readme.txt
=======================================================
    PhpGedView

    Version 4.1
    Copyright (c) 2005 to 2008 John Finlay and others

    This and other information can be found online at
    http://www.PhpGedView.net

    The installation instructions can also be found in the wiki at:
	http://wiki.phpgedview.net/en/index.php?title=Installation_Guide

    # $Id: readme.txt,v 1.3 2008/07/07 18:01:11 lsces Exp $
=======================================================

CONTENTS
     1.  LICENSE
     2.  INTRODUCTION
     3.  SYSTEM REQUIREMENTS
     4.  QUICK INSTALL
     5.  INSTALLATION
     6.  UPGRADING
     7.  UPDATING GEDCOMS
     8.  THEMES
     9.  MULTIMEDIA OBJECTS
    10.  RSS FEED
    11.  DATABASE TABLE LAYOUT
    12.  MANUAL CONFIGURATION
    13.  SECURITY
    14.  LANGUAGES
    15.  NON-STANDARD GEDCOM CODES
    16.  LANGUAGE EXTENSION FILES
    17.  MIGRATING FROM SQL TO INDEX MODE AND VICE VERSA
    18.  POSTNUKE AND PHPNUKE INTEGRATION
    19.  BACKUP

-------------------------------------------------------
LICENSE

PhpGedView: Genealogy Viewer
Copyright (C) 2002 to 2007  John Finlay and Others

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

See the file GPL.txt included with this software for more
detailed licensing information.

-------------------------------------------------------
INTRODUCTION

PhpGedView is a revolutionary genealogy program which allows you to view 
and edit your genealogy on your website.  PhpGedView has full editing 
capabilities, full privacy functions, and supports multimedia like photos 
and document images.  PhpGedView also simplifies the process of 
collaborating with others working on your family lines.  Your latest 
information is always on your web site and available for others to see.  
For more information and to see working demos, visit 
http://www.PhpGedView.net/

PhpGedView is Open Source software that has been produced by people from 
many countries freely donating their time and talents to the project.  All 
service, support, and future development is dependent on the time 
developers are willing to donate to the project, often at the expense of 
work, recreation, and family.  Beyond the few donations received from 
users, developers receive no compensation for the time they spend working 
on the project.  There is also no outside source of revenue to support the 
project.

-------------------------------------------------------
SYSTEM REQUIREMENTS

PhpGedView requires a web server with at least PHP v4.3.5 and around 20MB of
web space.  The default installations of PHP on most servers should provide
you with all of the PHP functionality you should need.

Database
    You will need at least 1 database and a username and password to access it. 
    PhpGedView primarily supports MySQL, but has also been tested and shown to 
    work with Postgresql, SQLite, and SQL-Server. The versions required for 
    each of these databases is outlined below: 
    MySQL 3.23+ 
    PostgresQL 8.0+ 
    SQLite available through PDO-SQLite which is included in PHP 5 
    MS SQL-Server 2003+ 

Web space
    At least 20MB of web space on the web server. You will need more than this 
    if you intend to store multimedia linked to individuals. 

To use the reporting engine, PHP needs to be compiled with XML support.  
This is compiled into PHP by default unless it is specifically disabled.  
See http://us3.php.net/manual/en/ref.xml.php

Some features of PhpGedView require the GD library and that PHP be compiled
with GD support.  Most precompiled versions of PHP include GD support.  If 
you are compiling PHP yourself you will need to configure it with the 
    --with-gd 
option.  See http://us3.php.net/manual/en/ref.image.php

The advanced calendar features of PhpGedView for converting Gregorian dates 
to the Hebrew or Jewish calendars require that PHP be compiled with the
    --enable-calendar 
configuration option.  See http://us2.php.net/manual/en/ref.calendar.php 
for more information.

For GEDCOMs larger than 2MB, you will likely need to research different
hosting options and find one that will accept a request to modify the
default memory and time limits built into PHP.  See the FAQ at
http://www.phpgedview.net/faq.php for more information about large GEDCOMs.

-------------------------------------------------------
QUICK INSTALLATION

These instructions can also be found in the wiki at:
http://wiki.phpgedview.net/en/index.php?title=Installation_Guide

Follow the instructions in this section to install PhpGedView if you are
already familiar with the program or are familiar with installing other PHP 
web applications.

 1.  Upload the files to your web server.
 2.  Copy config.dist to config.php
 3.  Set Write permissions on config.php and the "index" directory.  For 
     optimal security, you should move the "index" directory to a location 
     where it is not accessible from the Internet.
 4.  Point your browser to the directory where you uploaded your PhpGedView 
     files (for example, http://www.yourserver.com/PhpGedView/).
 5.  Enter your configuration settings.  If you moved the index directory, 
     be sure to specify the correct location to it on this page.  Save the 
     configuration parameters.
 6.  Enter the default administrator user.
 7.  Login as this user and upload your GEDCOM file.
 8.  Save the GEDCOM configuration settings.
 9.  Import the GEDCOM.

Optional Steps
10.  If you want to use the language editing features you will need to set 
     Write permissions for all of the files in the ./languages folder.
11.  If you want to upload media files using the Upload Media section of 
     the Admin menu then you need to set Write permissions for your ./media 
     and ./media/thumbs directories.
12.  If you want to edit your GEDCOM file online, the GEDCOM file must have
     Write permissions set for the PHP user.
13.  If you want to use the Backup feature of the Upgrade utility in 
     PhpGedView you will need to either set Write permission on the 
     PhpGedView folder itself or create a folder named "backup" with Write 
     permissions.  Write permissions for the PhpGedView folder can be 
     removed as soon as the backup folder is in place and has the 
     appropriate permissions.
14.  For security you should set the permissions back to Read-only when you 
     are done editing or uploading files.

-------------------------------------------------------
INSTALLATION

These instructions can also be found in the wiki at:
http://wiki.phpgedview.net/en/index.php?title=Installation_Guide

Follow these instructions if you are not familiar with PhpGedView or 
installing PHP applications.

*A.  Upload Program Files:
To install PhpGedView, unzip the compressed package and upload the files to 
a directory on your web server.  Rename the config.dist file to config.php.

If you have limited space on your server, you can save space in the following ways:
1.  Delete the themes from the themes folder that you do not plan to use.
2.  Delete some of the language files that you do not want.  English files 
    are named configure_help.en.php, countries.en.php, facts.en.php, 
    help_text.en.php and lang.en.php.  French files, for example, are named 
    with ".fr." in place of ".en.".  Hebrew files use ".he." in place of 
    ".en.", and so on.  
    The English language files cannot be deleted.  They are always loaded 
    before the files for the selected language are loaded.  This ensures 
    that all language variables are defined, and that the English version 
    will be used when a given variable is missing in the new language.
3.  Do not upload the entire "places" folder.  This folder contains maps for some
    countries.  It also contains text files containing state, county, and 
    place names.  Its purpose is to allow you to enter place names by 
    picking them from lists.    

For optimal security, you may want to move the "index" directory to a 
different location outside of your Internet accessible space.  You will 
specify the location of this directory during the online configuration.  
See the SECURITY section for more information.

*B.  Required File Permissions:
PhpGedView requires that Read permissions be set for all files in the
PhpGedView directory tree.  Some hosts also require Execute permissions
(chmod 755).  PhpGedView requires full Write permissions on the index 
directory (chmod 777 under most hosting configurations).  PhpGedView also 
requires that Write permissions (chmod 777) be set temporarily for the 
config.php file.

To help with the setting of permissions a file called setpermissions.php 
has been included with the project.  This file will attempt to set 777 
permissions to the config.php, ./index, and all of the files inside the 
./index directory.  Because host settings vary on the ability of PHP 
programs to set file permissions, you have to run this file manually.

If at any time you have trouble during configuration, check your 
permissions again.

There are some advanced features that require more Write permissions to be 
set.  If you want to use the language editing features you will need to set 
Write permissions for all of the files in the ./languages folder (chmod 
777).  If you want to upload media files using the Upload Media section of 
the Admin menu then you need to set Write permissions (chmod 777) for your 
./media and ./media/thumbs directories.  If you want to edit your GEDCOM 
file online, the GEDCOM file must have Write permissions set to the PHP 
user (chmod 777).

*C.  Configuration:
Next point your web browser to the PhpGedView folder
(for example, http://www.yourserver.com/PhpGedView/) to automatically 
begin the online configuration process.  Information about each of the 
configuration options can be found online by clicking on the question mark 
(?) near each label.

PhpGedView has support for importing your GEDCOMs into a PEAR:DB supported 
database like MySQL or PostgreSQL.  Currently, MySQL is the only fully tested 
database.  Using a database requires that an existing user, password, and 
database already exist.

You may reconfigure PhpGedView at any time by going to PhpGedView/admin.php 
and logging in as an administrator user and clicking on the "Configuration" 
link.

If you are having any problems setting up PhpGedView then you should run the 
sanity_check file. To do this you should type sanity_check.php into your URL 
(for example, http://www.yourserver.com/PhpGedView/sanity_check.php). If you are 
not able to view that page then you most likely don't have either the sanity_check 
file or you do not have PHP installed properly.

*D.  Create Admin User
After you click the Save button, you will be asked to create an 
administrator user and login as this user.  Then click on the link labelled 
"Click here to continue" where you will be taken to the "Manage GEDCOMs" 
area.  In the "Manage GEDCOMs" area you can add GEDCOMs to your site, edit 
the GEDCOM configuration, edit Privacy settings, and import the GEDCOM into 
the data store.

*E.  Add GEDCOM file
To add GEDCOM files to the system, you can upload your GEDCOM file using 
the "Upload GEDCOM" option from the Admin menu.  All files uploaded using 
the "Upload GEDCOM" page are saved in your index directory.  You can also 
upload your GEDCOM manually using FTP or any other file upload method.  
Most hosts limit the size of files that can be uploaded from a web form for 
security reasons, so you may be forced to use a manual method.  You may 
also upload your GEDCOM in ZIP format, either manually or using the 
"Upload GEDCOM" option.  Make sure to enter the filename of the ZIP file.  
PhpGedView will automatically unpack the ZIP file and use the GEDCOM file 
inside it.  Be sure to create the ZIP file to contain only one GEDCOM file.

*F.  Set GEDCOM Configuration Settings
After uploading your GEDCOM, you will be asked to set the configuration
parameters for it.  There are too many parameters to list all of their 
options in this document.  Please use the online Help documentation to 
guide you through the configuration process.

*G.  Validate GEDCOM
After you save the GEDCOM configuration PhpGedView will validate your 
GEDCOM and automatically fix any errors that it can.  If any errors found 
in the GEDCOM require user input, you will be prompted to choose how to 
proceed.  Again use the online Help ? for more information.

*H.  Import GEDCOM
You are almost done.  This is the final step before you can begin viewing 
your data.  After validating the GEDCOM and fixing any errors, you will 
need to import the GEDCOM into the data store.  During the Import you will 
see a lot of processing statistics printed on the screen.  If the Import 
completed successfully you will see a blue "Import Complete" message.  
Everything is now set up and you can begin using PhpGedView with your 
GEDCOM.

*I.  Deleting GEDCOMs
You may delete GEDCOMs from your site from the "Manage GEDCOMs" area.
Deleting a GEDCOM from the site will remove it from the database but will 
not delete the original GEDCOM file that you imported.  It will also not
delete any of the cache or privacy files related to this GEDCOM.  These 
retained files, which are no longer required by PhpGedView, are all located 
in the "index" directory.

*J.  Reset config.php Permissions
For security you should set the permissions of config.php back to Read-only
(chmod 755) when you have finished configuring for the first time.  Write
permissions will only need to be set for config.php when you use the
Admin->Configuration link.  Everything else will be stored in the index
directory.

*K.  Custom Themes
You can customize the look and feel of PhpGedView by modifying one of the
provided themes.  See the THEMES section of this readme file for more
information.

*L.  HTTP Compression
Pages generated by PhpGedView can be large and use up a lot of bandwidth.
Compression of the data between the server and browser using GZip 
compression can compress the bandwidth by up to 90% (usually 80% - 90% for 
PhpGedView that were tested).  Add the following 2 lines to your php.ini file:
    zlib.output_compression On
    zlib.output_compression_level 5

If you have no access to the php.ini file and you are using Apache, create a 
blank file named .htaccess (including the dot) and add the following lines to 
that file  (or add them to an existing .htaccess file and upload the file to 
your PhpGedView directory.
    php_flag zlib.output_compression On
    php_value zlib.output_compression_level 5

Some hosts do not allow adding this through .htaccess files, but they may 
allow you to create a partial php.ini file in your phpGedView directory. To
this file you would add the same two lines from the php.ini file above:
    zlib.output_compression On
    zlib.output_compression_level 5

Note: If your host is using mod_gzip or an other compression method, using 
this technique can cause problems.  Compression will have no effect on 
browsers that do not support it.  You can test the compression at
http://leknor.com/code/gziped.php

If you need help or support visit  http://www.PhpGedView.net/support.php

-------------------------------------------------------
UPGRADING

UPGRADING 4.1 to 4.1.x

1. To upgrade from 4.1 to 4.1.1 or any other 4.1.x version, simply replace 
   the files on the server with the new files.
2. Some of the internal structures have changed, so you should also delete
   your cache files.  This can be done from the "Customize Welcome Page" link
   or by manually deleting the files from the "index" directory. 


UPGRADING 4.0.x to 4.1.x

Use the following steps to upgrade to v4.1.  These steps assume that you 
are familiar with PhpGedView and have successfully installed it before.

Version 4.x no longer includes support for index files.  If you do not have 
PHP 5 with built-in SQLite support or a database account then you should not 
upgrade to PhpGedView 4.x.  Maintenance releases will continue for the 3.3.x 
version.

**Note to SQLite users: Unfortunately SQLite does not support the ALTER TABLE
  SQL command.  This means that we must first drop the tables and recrate
  them.  It is highly reccomended that you run the backup before upgrading.

1.  Upload the new 4.1 files to your server replacing the old files with the 
	new files.  Do not replace the config.php file or the index directory.
2.  In order to create the data required for the new version, you will
    have to re-import your GEDCOM. Please see the section on UPDATING
    GEDCOMS below.
3.  You can now use your upgraded site.
4.  If you are using a customized theme you will need to update your theme
    with new stylesheets and variables.  An excellent tool that can help 
    you to merge themes is the WinMerge project 
    http://winmerge.sourceforge.net/
    
-------------------------------------------------------
UPDATING GEDCOMS

When you change your genealogy data outside of PhpGedView, it is not 
necessary to delete your GEDCOMs from PhpGedView and start over.  Follow 
these steps to update a GEDCOM that has already been imported:

1.  The first step is to replace your old GEDCOM on the site with your new
    GEDCOM.  You can do this using FTP, or by going to the "Upload GEDCOM" 
    page and uploading a new GEDCOM with the same filename as the old one.  
    Please remember that file names are case sensitive.
2.  Re-import the GEDCOM file by going to 
    Admin->Manage GEDCOMs->Import GEDCOM.  The GEDCOM will be validated 
    again before importing.
3.  The Import script will detect that the GEDCOM has already been imported 
    and will ask if you want to replace the old data.  Click the "Yes" 
    button.
4.  You will again see the Import statistics and the Import Complete 
    message at the bottom of the page when the Import is complete.

If you use a ZIP file to upload the GEDCOM, the only way to do it is by 
either using the "Upload GEDCOM" option, or the "Add GEDCOM" option.  The 

changelog.txt
=====================================================================
	Change Log
	phpGedView

	This and other information can be found online at
	http://www.phpgedview.net

	# $Id$
=====================================================================


Version 4.1.5
 - Fixed: [ 1952284 ] Unknown names print without the (unknown) text (fisharebest)
 - Fixed: [ 1952073 ] SVN 2918 - STATS, bad SURN numbers (fisharebest)
 - Added: Standardised interface to CMS applications such as joomla, drupal, postnuke, etc. (fisharebest)
 - Fixed: [ 1951591 ] Error in Hebrew Date Editing (fisharebest)
 - Fixed: [ 1951335 ] Privacy Warning on View GEDCOM for Media Even for Admin (fisharebest)
 - Fixed: [ 1950986 ] Remote Link - Person to Person, SVN 2899 (fisharebest)
 - Fixed: [ 1947687 ] [4.1.4] Edit link in Quick Update does not work (fisharebest)
 - Fixed: [ 1946362 ] SVN 2867 - Quick Update Links to wrong GEDCOM (fisharebest)
 - Fixed: [ 1950235 ] SVN2887- Search Bug (fisharebest)
 - Fixed: [ 1950232 ] PGSQL: Syntax-error in functions_db.php (fisharebest)
 - Fixed: FACT/TYPE not being checked for privacy rules when editing raw gedcom (fisharebest)
 - Fixed: [ 1933760 ] SVN 2799 use of function id_type (fisharebest)
 - Fixed: Remove dependencies on record prefixes (fisharebest)
 - Fixed: Extra checks against reusing record IDs (fisharebest)
 - Fixed: [ 1949549 ] GEDCOM Stats block: incorrect block title (canajun2eh)
 - Fixed: Use stats class to calculate stats (fisharebest)
 - Fixed: Stats block counting records from other gedcoms (fisharebest)
 - Fixed: [ 1942593 ] Relationship Chart nth Cousin Calculation Issue (canajun2eh)
 - Fixed: Temp Fix for Lightbox giving errors when Googlemaps not installed (windmillway)
 - Fixed: [ 1944250 ] placelist.php Level Issue On Language Change For Some Themes (canajun2eh)
 - Added: Add "Painting" and "Other" to supported sortable media types on Lightbox-Album page(windmillway)
 - Fixed [ 1941544 ] Yahrzeiten Block in IE7 gives an Error (opus27)
 - Added: Add "Painting" and "Other" to supported media types (canajun2eh)
 - Fixed: [ 1943975 ] uploadmedia.php : undefined index "folder1" (canajun2eh)
 - Fixed: [ 1941658 ] Account Approval Email not sent to Admin (fisharebest)
 - Fixed: Move "maternal-death" style to stylesheet (fisharebest)
 - Added: Highligh fathers who die before their child is born (fisharebest)
 - Added: Option to show estimated birth/death dates on indi lists (fisharebest)
 - Fixed [ 1939774 ] MyGedView Portal Menu Showing even for Non-Logged In Users (canajun2eh)
 - Fixed: wrong gedcom record updated when syncing email changes to gedcom (fisharebest)
 - [ 1921974 ] SVN2756 Wrong names in search list for several GEDCOMs (fisharebest)
 - Fixed: weak randomly-generated passwords (fisharebest)
 - Fixed Logout sometimes goes to PGV site in SourceForge (canajun2eh)

Version 4.1.4 2008-04-04
 - Added Login block changes to Logout block once user is logged in (canajun2eh)
 - Fixed Welcome page in IE: blocks on right side are all fixed height (canajun2eh)
 - Fixed [1924688] SVN 2755 V4.1.4 Incorrect display of Indi page with IE (fisharebest)
 - Fixed [1912085] Syntax Error in downloadbackup.php (fisharebest)
 - Fixed [1907585] RSS Error
 - Fixed [1906576] Indi/Fam lists ignore event privacy settings (fisharebest)
 - Fixed [1900498] SVN 2610 - changes effecting Joomla Bridge (fisharebest)
 - Fixed [1718333] New user verification fails when hash ends with dot (fisharebest)
 - Fixed [1915244] SVN 2719 Caused a Source Citation Display issue on indi page (fisharebest)
 - Fixed [1912716] V4.1.4 Random picture block size (fisharebest)
 - Fixed [1912525] V4.1.4 User edit screen not displaying correctly (fisharebest)
 - Fixed [1910328] PHP Errors in useradmin.php (fisharebest)
 - Fixed [1908247] (4.1.4) Backup error, advanced-html block (fisharebest)
 - Fixed [1907581] GBR_England.txt and ENG.htm (fisharebest)
 - Fixed [1906629] Parent's age at child's birth wrong (fisharebest)
 - Fixed [1906444] SVN 2662 V4.1.4 errors on Review Changes block (fisharebest)
 - Fixed [1905569] ERROR 8: unserialize(): Argument is not a string(fisharebest)
 - Fixed [1905512] SVN 2657: Logout returns to MyGedview page as logged-in user (fisharebest)
 - Fixed [1903888] [SVN 2640] V4.2 tempuser variable not defined (fisharebest)
 - Fixed [1903621] [SVN 2639] PHP Errors in useradmin.php (fisharebest)
 - Fixed [1902458] user edits return to wrong GEDCOM (fisharebest)
 - Fixed [1901932] PHP 5.2.5 upgrade has caused errors (fisharebest)
 - Fixed [1901416] Visitors see the Logged-In User List (fisharebest)
 - Fixed [1901229] SVN 2612 Unable to see site in version 4.2 (fisharebest)
 - Fixed [1901099] 4.1.4 svn 2611 Can't authenticate new user (fisharebest)
 - Fixed [1896595] SVN 2576: Logout returns to MyGedview page as logged-in user (fisharebest)
 - Fixed [1895530] SVN 2559: undefined index errors in useradmin.php (fisharebest)
 - Fixed [1891170] Recognize Me - New Portal loading issues-site DOWN (fisharebest)
 - Fixed [1886435] Incorrect format for _UID fields (fisharebest)
 - Fixed [1882371] class_stats.php - date error with @#DGREGORIAN@ (fisharebest)
 - Fixed [1877920] Non Latin1 letters in user definition (fisharebest)
 - Fixed [1871905] (4.1.3) SQLite Error (fisharebest)
 - Fixed [1843093] Todays Events - Linking SVN 2360 (fisharebest)
 - Fixed [1842251] 'Edit Name' adds a slash to the end (fisharebest)
 - Fixed [1838108] 4.2 SVN (2277) useradmin.php PHP Error (fisharebest)
 - Fixed [1835401] User Comment - New Line - Wrong Print (fisharebest)
 - Fixed [1833111] Userfriendly Date editing on the Finnish page (fisharebest)
 - Fixed [1833098] Userfriendly Hebrew Date Editing - Error (fisharebest)
 - Fixed [1832031] Jewish Calendar days on Hebrew page (fisharebest)
 - Fixed [1829834] Yahrzeiten Block sorting of Anniversaries is wrong (fisharebest)
 - Fixed [1829822] Hebrew Death shown with (1 day after death) (fisharebest)
 - Fixed [1827152] Upcoming Events Block - Sorting (fisharebest)

rss.php
<?php
/**
 * Outputs an ATOM or RSS feed of information, mostly based on the information available
 * in the index page.
 *
 * phpGedView: Genealogy Viewer
 * Copyright (C) 2002 to 2004  John Finlay and Others
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 *
 * $Id: rss.php,v 1.7 2008/07/07 18:01:11 lsces Exp $
 * @package PhpGedView
 * @subpackage RSS
 * @TODO add Basic HTTP authentication to allow RSS aggregators to "log on"
 */

if (isset($_SESSION["CLANGUAGE"])) $oldlang = $_SESSION["CLANGUAGE"];
else $oldlang = "english";
if (!empty($lang)) {
	$changelanguage = "yes";
	$NEWLANGUAGE = $lang;
}
require("includes/feedcreator.class.php");
require("includes/functions_rss.php");
require("config.php");
require("includes/index_cache.php");

$feedCacheName = "fullFeed";

/*if (!empty($auth)){
	if($auth=="basic"){
		basicHTTPAuthenticateUser();
	}
}*/

// valid format strings are: RSS0.91, RSS1.0, RSS2.0, MBOX, OPML, ATOM, ATOM1.0, ATOM0.3, HTML, JS
if (empty($rssStyle)){
	if (!empty($RSS_FORMAT)) $rssStyle = $RSS_FORMAT;
	else $rssStyle = "ATOM";	// Unless configured otherwise, default to ATOM
}

if (!isset($_SERVER['QUERY_STRING'])){
	$_SERVER['QUERY_STRING'] = "lang=".$LANGUAGE;
}

$printTodays = false;
$printUpcoming = false;
$printGedcomStats = false;
$printGedcomNews = false;
$printTop10Surnames = false;
$printRecentChanges = false;
$printRandomMedia = false;

if(!empty($module)){
	$feedCacheName = $module . "Feed";
	if($module == "today"){
		$printTodays = true;
	} else if($module == "upcoming"){
		$printUpcoming = true;
	} else if($module == "gedcomStats"){
		$printGedcomStats = true;
	} else if($module == "gedcomNews"){
		$printGedcomNews = true;
	} else if($module == "top10Surnames"){
		$printTop10Surnames = true;
	} else if($module == "recentChanges"){
		$printRecentChanges = true;
	} else if($module == "randomMedia"){
		$printRandomMedia = true;
	}
}

// Build the array to control caching
$cacheControl = array();
$cacheControl[0] = $feedCacheName;
$cacheControl[1] = array("cache"=>1);
if (!empty($module) && $module=="randomMedia") $cacheControl[1]["cache"] = 0;

if(!loadCachedBlock($cacheControl, $rssStyle)){
	$author=getUserFullName($CONTACT_EMAIL);

	$feed = new UniversalFeedCreator();
	$feed->generator = "http://www.phpgedview.net v" . $VERSION . " " . $VERSION_RELEASE;
	$feed->title = $gGedcom[$GEDCOM]["title"];
	$feed->language = $lang_short_cut[$LANGUAGE]; //$lang_langcode[$LANGUAGE];
	$feed->descriptionHtmlSyndicated = true;
	//$feed->descriptionTruncSize = 500; // does not make sense to truncate HTML since it will result in unpredictable output
	$feed->link = $SERVER_URL;
	$syndURL = $SERVER_URL."rss.php?".$_SERVER['QUERY_STRING'];
	$syndURL = preg_replace("/&/", "&amp;", $syndURL);
	$feed->syndicationURL = $syndURL;

	$feedDesc = str_replace("#GEDCOM_TITLE#", $gGedcom[$GEDCOM]["title"], $pgv_lang["rss_descr"]);
	$feed->description = $feedDesc;
	$feed->copyright = $author . " (c) " . date("Y");
	$feed->category="genealogy";

	$image = new FeedImage();
	$image->title = $pgv_lang["rss_logo_descr"];
	$image->url = $SERVER_URL."images/gedview.gif";
	$image->link = "http://www.phpgedview.net";
	$image->description = $pgv_lang["rss_logo_descr"];
	$image->descriptionHtmlSyndicated = true;
	//$feed->descriptionTruncSize = 500; // does not make sense to truncate HTML since it will result in unpredictable output
	$feed->image = $image;



	if($ENABLE_RSS) {
		// basic auth is broken in all apps besides browsers, so I am disabling it for now.
		/*if (empty($auth) || $auth != "basic"){
			if(!PGV_USER_ID){ //not logged in.
				$item = new FeedItem();
				$item->title = $pgv_lang["login"];
				$authURL= $syndURL . "&auth=basic";
				$item->link = $authURL;
				//$item->guid = $item->link;
				$authDesc = str_replace("#AUTH_URL#", $authURL, $pgv_lang["feed_login"]);
				$item->description = $authDesc;
				$item->descriptionHtmlSyndicated = true; //optional
				$item->date = time();
				$item->source = $SERVER_URL;
				$item->author = $author;
				$item->authorURL = $feed->link;
				$item->category = $pgv_lang["genealogy"];
				$feed->addItem($item);
			}
		}*/

		// determine if to show parts of feed based on their exsistance in the blocks on index.php
		$blocks=  getBlocks($GEDCOM);
		$main = $blocks["main"];

		if(empty($module)) {
			if (count($main)==0) {
				$printGedcomStats = true;
				$printGedcomNews = true;
			} else {
				foreach($main as $mname => $value){
					$PGV_BLOCKS[$value[0]]['config'] = $value[1]; //set the config needed by functions_rss
					if($value[0] == "print_todays_events"){
						$printTodays = true;
					} else if($value[0] == "print_upcoming_events"){
						$printUpcoming = true;
					} else if($value[0] == "print_gedcom_stats"){
						$printGedcomStats = true;
					} else if($value[0] == "print_gedcom_news"){
						$printGedcomNews = true;
					} else if($value[0] == "print_block_name_top10"){
						$printTop10Surnames = true;
					} else if($value[0] == "print_recent_changes"){
						$printRecentChanges = true;
					} else if($value[0] == "print_random_media"){
						$printRandomMedia = true;
					}
				}
			}
			$right = $blocks["right"];
			if (count($right)==0) {
				$printTodays = true;
			} else {
				foreach($right as $mname => $value){
					$PGV_BLOCKS[$value[0]]['config'] = $value[1]; //set the config needed by functions_rss
					if($value[0] == "print_todays_events"){
						$printTodays = true;
					} else if($value[0] == "print_upcoming_events"){
						$printUpcoming = true;
					} else if($value[0] == "print_gedcom_stats"){
						$printGedcomStats = true;
					} else if($value[0] == "print_gedcom_news"){
						$printGedcomNews = true;
					} else if($value[0] == "print_block_name_top10"){
						$printTop10Surnames = true;
					} else if($value[0] == "print_recent_changes"){
						$printRecentChanges = true;
					} else if($value[0] == "print_random_media"){
						$printRandomMedia = true;
					}
				}
			}
		}

		if($printTodays){
			$todaysEvents = getTodaysEvents();
			if (! empty($todaysEvents[2])) {
				$item = new FeedItem();
				$item->title = $todaysEvents[0];
				$item->link = $SERVER_URL. "calendar.php?action=today";
				$item->description = $todaysEvents[2];
				$item->descriptionHtmlSyndicated = true;
				$item->date = $todaysEvents[1];
				$item->source = $SERVER_URL;
				$item->author = $author;
				$item->authorURL = $feed->link;
				$item->category = $pgv_lang["genealogy"];
				$feed->addItem($item);
			}
		}

		if($printUpcoming){
			$upcomingEvent = getUpcomingEvents();
			if (! empty($upcomingEvent[2])) {
				$item = new FeedItem();
				$item->title = $upcomingEvent[0];
				$item->link = $SERVER_URL. "calendar.php?action=calendar";
				$item->description = $upcomingEvent[2];
				$item->descriptionHtmlSyndicated = true;
				$item->date = $upcomingEvent[1];
				$item->source = $SERVER_URL;
				$item->author = $author;
				$item->authorURL = $feed->link;
				$item->category = $pgv_lang["genealogy"];
				$feed->addItem($item);
			}
		}

		if($printGedcomStats){
			$gGedcomtats = getGedcomStats();
			if (! empty($gGedcomtats[2])) {
				$item = new FeedItem();
				$item->title = $gGedcomtats[0];
				$item->link = $SERVER_URL. "index.php?ctype=gedcom#gedcom_stats";
				$item->description = $gGedcomtats[2];
				$item->descriptionHtmlSyndicated = true;
				if (! empty($gGedcomtats[1])) {
					$item->date = $gGedcomtats[1];
				}
				$item->source = $SERVER_URL;
				$item->author = $author;
				$item->authorURL = $feed->link;
				$item->category = $pgv_lang["genealogy"];
				$feed->addItem($item);
			}
		}

		if($printTop10Surnames){
			$top10 = getTop10Surnames();
			if (! empty($top10[2])) {
				$item = new FeedItem();
				$item->title = $top10[0];
				$item->link = $SERVER_URL. "indilist.php";
				$item->description = $top10[2];
				$item->descriptionHtmlSyndicated = true;
				if (! empty($top10[1])) {
					$item->date = $top10[1];
				}
				$item->source = $SERVER_URL;
				$item->author = $author;
				$item->authorURL = $feed->link;
				$item->category = $pgv_lang["genealogy"];
				$feed->addItem($item);
			}
		}

		if($printGedcomNews){
			$gedcomNews = getGedcomNews();

			$numElements = count($gedcomNews); //number of news items
			for($i=0; $i < $numElements; $i++) {
				$newsItem = $gedcomNews[$i];
				if (! empty($newsItem[1])) {
					$item = new FeedItem();
					$item->title = $newsItem[0];
					$item->link = $SERVER_URL . "index.php?ctype=gedcom#" . $newsItem[3];
					$item->description = $newsItem[2];
					$item->descriptionHtmlSyndicated = true;
					$item->date = $newsItem[1];
					$item->source = $SERVER_URL ;
					$item->author = $author;
					$item->authorURL = $feed->link;
					$item->category="genealogy";
					$feed->addItem($item);
				}
			}
		}

		if($printRecentChanges){
			$recentChanges= getRecentChanges();
			if (! empty($recentChanges[2])) {
				$item = new FeedItem();
				$item->title = $recentChanges[0];
				$item->link = $SERVER_URL. "index.php?ctype=gedcom#recent_changes";
				$item->description = $recentChanges[2];
				$item->descriptionHtmlSyndicated = true;

				if (! empty($recentChanges[1])) {
					$item->date = $recentChanges[1];
				}
				$item->source = $SERVER_URL;
				$item->author = $author;
				$item->authorURL = $feed->link;
				$item->category = $pgv_lang["genealogy"];
				$feed->addItem($item);
			}
		}

		if($printRandomMedia){
			$randomMedia= getRandomMedia();
			if (! empty($randomMedia[2])) {
				$item = new FeedItem();
				$item->title = $randomMedia[0];
				$item->link = $SERVER_URL. "medialist.php";
				$item->description = $randomMedia[2];
				$item->descriptionHtmlSyndicated = true;

				if (! empty($randomMedia[1])) {
					$item->date = $randomMedia[1];
				}
				$item->source = $SERVER_URL;
				$item->author = $author;
				$item->authorURL = $feed->link;
				$item->category = $pgv_lang["genealogy"];
				$item->enclosure = new EnclosureItem();
				$item->enclosure->url = $SERVER_URL . $randomMedia[3];
				$item->enclosure->type = $randomMedia[4];
				$item->enclosure->length = $randomMedia[5];
				$item->enclosure->title = $randomMedia[6];

				$feed->addItem($item);
			}
		}
	} else {
		$item = new FeedItem();
		$item->title = $pgv_lang["no_feed_title"];
		$item->link = $SERVER_URL. "index.php";
		$item->description = $pgv_lang["no_feed"];
		$item->date = time();
		$item->source = $SERVER_URL;
		$item->author = $author;
		$item->authorURL = $feed->link;
		$item->category = $pgv_lang["genealogy"];
		$feed->addItem($item);
	}

	//$feed->outputFeed($rssStyle);

	ob_start();
	$feed->outputFeed($rssStyle);
	$content = ob_get_contents();
	saveCachedBlock($cacheControl, $rssStyle, $content);
	ob_end_flush();
}
 //-- preserve the old language by storing it back in the session
$_SESSION['CLANGUAGE'] = $oldlang;

?>
